File: CS/SubmitMultipleValues/Controllers/Test2Controller.cs
using DevExpress.Web.Mvc;
using SubmitMultipleValues.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace SubmitMultipleValues.Controllers
{
    public class Test2Controller : Controller
    {     
        public ActionResult Index()
        {
            return View(new MainModel());
        }
        public ActionResult GridViewPartial()
        {
            return PartialView(BatchEditRepository.GridData);
        }
        public ActionResult BatchUpdatePartial(MVCxGridViewBatchUpdateValues<GridDataItem, int> batchValues)
        {                      
            foreach (var item in batchValues.Insert)
            {
                if (batchValues.IsValid(item))
                    BatchEditRepository.InsertNewItem(item, batchValues);
                else                
                    batchValues.SetErrorText(item, "Correct validation errors");                
            }
            foreach (var item in batchValues.Update)
            {
                if (batchValues.IsValid(item))
                    BatchEditRepository.UpdateItem(item, batchValues);
                else
                    batchValues.SetErrorText(item, "Correct validation errors");
            }
            foreach (var itemKey in batchValues.DeleteKeys)
            {
                BatchEditRepository.DeleteItem(itemKey, batchValues);
            }
            ViewData["ErrorFlag"] = batchValues.EditorErrors.Count > 0;
            return PartialView("GridViewPartial", BatchEditRepository.GridData);
        }
        [HttpPost]
        public ActionResult PostModel(MainModel m){
            if (ModelState.IsValid)
            {
                /*get Batch Edit data from a temp repository and save it to a data base with a main model */
                return View("Success");
            }
            else
                return View("Index");
        }
    }
}

File: VB/SubmitMultipleValues/Controllers/Test2Controller.vb
Imports Microsoft.VisualBasic
Imports DevExpress.Web.Mvc
Imports SubmitMultipleValues.Models
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc

Namespace SubmitMultipleValues.Controllers
	Public Class Test2Controller
		Inherits Controller
		Public Function Index() As ActionResult
			Return View(New MainModel())
		End Function
		Public Function GridViewPartial() As ActionResult
			Return PartialView(BatchEditRepository.GridData)
		End Function
		Public Function BatchUpdatePartial(ByVal batchValues As MVCxGridViewBatchUpdateValues(Of GridDataItem, Integer)) As ActionResult
			For Each item In batchValues.Insert
				If batchValues.IsValid(item) Then
					BatchEditRepository.InsertNewItem(item, batchValues)
				Else
					batchValues.SetErrorText(item, "Correct validation errors")
				End If
			Next item
			For Each item In batchValues.Update
				If batchValues.IsValid(item) Then
					BatchEditRepository.UpdateItem(item, batchValues)
				Else
					batchValues.SetErrorText(item, "Correct validation errors")
				End If
			Next item
			For Each itemKey In batchValues.DeleteKeys
				BatchEditRepository.DeleteItem(itemKey, batchValues)
			Next itemKey
			ViewData("ErrorFlag") = batchValues.EditorErrors.Count > 0
			Return PartialView("GridViewPartial", BatchEditRepository.GridData)
		End Function
		<HttpPost> _
		Public Function PostModel(ByVal m As MainModel) As ActionResult
			If ModelState.IsValid Then
				'get Batch Edit data from a temp repository and save it to a data base with a main model 
				Return View("Success")
			Else
				Return View("Index")
			End If
		End Function
	End Class
End Namespace
